src/services/VersionCheckerService.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Symbol Contributors 2021 (https://nem.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+import { appConfig } from '@/config';
+import fetch from 'node-fetch';
+
+/**
+ * Request github api for latest release
+ * @internal
+ * @return {Promise<LatestVersionObject>}
+ */
+const request = async (): Promise<LatestVersionObject> => {
+    let versionTag;
+    let downloadUrl;
+    await fetch(appConfig.repositoryDataUrl)
+        .then((response) => response.json()) //Converting the response to a JSON object
+        .then((data) => {
+            versionTag = data.tag_name;
+            downloadUrl = data.html_url;
+        })
+        .catch((error) => console.error(error));
+    return { versionTag, downloadUrl };
```

Comment:
The function should return `Promise<{versionTag: string, downloadUrl: string}` 
but in case of error it will be `Promise<{versionTag: undefined, downloadUrl: undefined}`.

Might it be better to let the error propagation and remove `.catch`?

Reply:
I'd better leave the catch to see what's going on if something and handle the other case of `versionTag` might be undefined.